### PR TITLE
Bugfix/form validation

### PR DESCRIPTION
**Pull request by ntsummers1**
# Title

Bugfix/form validation

# Summary

This PR adds validation to the application form. It does a pretty big re-write of the app form to use [Formik](https://formik.org/), a really nice form generation tool. It enabled error messaging with yup validation to be really easy peasy. That being said, this will likely need to be ported so that the new filters can use it. Might want to get https://github.com/coforma/swift-tech-challenge/pull/169 in first.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How To Test

# Checklist:

- [x] My code follows the style guidelines of this project and has no linting errors
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, including any applicable ADRs
- [x] My changes generate no new warnings
- [x] I have added tests that fail without these changes
- [x] New and existing tests (unit, integration, accessibility) pass locally
- [ ] Documentation updated
- [ ] If there are security concerns they are addressed or ticketed after being discussed

### CODE REVIEW

File: src/app/components/form/AppForm.tsx [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
+import NotFound from "../../not-found";
+import { TextField } from "./TextField";
 import Link from "next/link";
+import { saveApplication } from "../../utils/applications";
 import { useRouter } from "next/navigation";
-// utils
-import { getInstitutionApplication } from "@/src/app/utils/institutions";
-import { saveApplication } from "@/src/app/utils/applications";
+import { email, number, numberOptional, text } from "../../schemas/schemas";
+import { TextArea } from "./TextArea";
```
Comment (gmrabian) [inline src/app/components/form/AppForm.tsx:21]: nit: can combine some of the imports from the component index file
Spinner, TextArea, TextField

Comment (gmrabian) [reply src/app/components/form/AppForm.tsx:21]: and if you do end up doing that, would be nice to organize these imports with comments like elsewhere

File: src/app/testing/jest/frontend/Apply.test.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ describe("Test Application Page", () => {
     expect(screen.getByText("Phone number")).toBeVisible();
     expect(screen.getByText("Email")).toBeVisible();
     expect(screen.getByText("Writing")).toBeVisible();
-    expect(screen.getByText("Essay Question 1")).toBeVisible();
+    // expect(screen.getByText("Essay Question 1")).toBeVisible();
```
Comment (gmrabian) [inline src/app/testing/jest/frontend/Apply.test.tsx:44]: remove if not needed

**Review by gmrabian — COMMENTED** [review]
Overall looks good to me. Do Formik and react-hook-form play nicely? How does USWDSForm help as well?

File: src/app/testing/jest/frontend/Apply.test.tsx (hunk 2 above)
Comment (ntsummers1) [reply src/app/testing/jest/frontend/Apply.test.tsx:44]: Is needed. Good catch :D

File: src/app/components/form/AppForm.tsx (hunk 1 above)
Comment (ntsummers1) [reply src/app/components/form/AppForm.tsx]: Fixed up :D

**Comment by ntsummers1** [comment]
> Overall looks good to me. Do Formik and react-hook-form play nicely? How does USWDSForm help as well?

So Formik is a replacement for react-hook-form. We should use one or the other, and Formik comes with Yup integration so you don't need a 3rd party to hook up react-hook-form to yup. Also, its much lighter which is a plus.

USWDS form is a form wrapper, so you can put whatever fields you want inside of it and it'll contain it. I've used it here so that it can wrap and control the modal form's filters as well as the application form all using the same code. Pretty slick!

**Review by braxex — DISMISSED** [review]

**Review by braxex — APPROVED** [review]

**Pull request merged**